tests/models/test_default_evaluator.py

```diff
@@ -65,20 +68,120 @@ def assert_dict_equal(d1, d2, rtol):
         assert np.isclose(d1[k], d2[k], rtol=rtol)
 
 
-def test_regressor_evaluation(linear_regressor_model_uri, diabetes_dataset):
+def evaluate_model_helper(
+    model,
+    baseline_model,
+    data,
+    targets,
+    model_type: str,
+    dataset_name=None,
+    evaluators=None,
+    evaluator_config=None,
+    is_baseline_model=None,
+):
+
+    """
+    Helper function for testing MLflow.evaluate
+    To test if evaluation for baseline model does not log metrics and artifacts;
+    we set "is_baseline_model" to true for all evaluator_config
+    """
+    if is_baseline_model:
+        if not evaluator_config:
+            evaluator_config = {"is_baseline_model": True}
+        elif not evaluators or evaluators == "default":
+            evaluator_config.update({"is_baseline_model": True})
+        else:
+            for config in evaluator_config.values():
+                config.update({"is_baseline_model": True})
+
+    return evaluate(
+        model=baseline_model if is_baseline_model else model,
+        data=data,
+        model_type=model_type,
+        targets=targets,
+        dataset_name=dataset_name,
+        evaluators=evaluators,
+        evaluator_config=evaluator_config,
+    )
+
+
+def metrics_check_helper(
+    logged_metrics, result_metrics, expected_metrics, dataset_log_key, test_is_baseline_model_flag
+):
+    """
+    Helper function for unit tests for evaluation and validation for check metrics:
+        - if test_is_baseline_model_flag:
+            Check if no metric is logged.
+        - Othwerise
+            Check if logged metrics are expected.
+        Check if returned metrics are expected.
+    """
+    if test_is_baseline_model_flag:
+        assert logged_metrics == {}
+    for metric_key in expected_metrics:
+        if not test_is_baseline_model_flag:
+            assert np.isclose(
+                expected_metrics[metric_key],
+                logged_metrics[metric_key + dataset_log_key],
+                rtol=1e-3,
+            )
+        assert np.isclose(expected_metrics[metric_key], result_metrics[metric_key], rtol=1e-3)
+
+
+def artifacts_check_helper(
```

Comment:
Are there more descriptive names we can use for these methods that would make it obvious what we're checking? I also think it's acceptable to split these into multiple methods for readability.

Reply:
Renamed them and split them into two functions for better readability.